torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -593,6 +593,41 @@ std::vector<ComputationClient::ComputationPtr> PjRtComputationClient::Compile(
   return computations;
 }
 
+std::string PjRtComputationClient::SerializeComputation(
+    ComputationPtr computation) {
+  const PjRtComputation& pjrt_computation =
+      dynamic_cast<const PjRtComputation&>(*computation);
+
+  return ConsumeValue(pjrt_computation.executable->SerializeExecutable());
+}
+
+ComputationClient::ComputationPtr PjRtComputationClient::DeserializeComputation(
+    std::string& serialized) {
+  auto executable_or = client_->DeserializeExecutable(serialized, std::nullopt);
+  if (!executable_or.ok()) {
+    TF_LOG(WARNING) << "Failed to deserialize executable: "
+                    << executable_or.status();
+    return nullptr;
+  }
+  auto executable = std::move(*executable_or);
+
+  auto hlo_modules = executable->GetHloModules();
+  if (!hlo_modules.ok()) {
+    TF_LOG(WARNING)
+        << "Failed to retrieve HLO modules from deserialized executable";
+    return nullptr;
+  }
+  XLA_CHECK(hlo_modules->size() == 1)
+      << "Only a single module is supported for persistent computation "
+         "caching. Please unset the XLA_PERSISTENT_CACHE_PATH "
+         "variable to disable persistent caching.";
+  xla::XlaComputation computation((*hlo_modules)[0]->ToProto());
+
+  std::vector<std::string> devices = {GetDefaultDevice()};
```

Comment:
@JackCaoG I'm not sure what this should be for SPMD mode. Since it's in the runtime layer, this will result in `TPU:0` with SPMD, but should it be `SPMD:0`? It doesn't seem to be causing any issues as is.

Reply:
Hmm I guess for SPMD we only load it once in one process? I don't remember exactually what I did but I am wondering if we compile the program normally without cache, would device return in `PjRtComputation` has `TPU:0` or `SPMD:0`?

Reply:
Looks like `SPMD:0`, I'll change this.